include/oboe/FullDuplexStream.h

```diff
@@ -27,16 +27,21 @@ namespace oboe {
 /**
  * FullDuplexStream can be used to synchronize an input and output stream.
  *
- * For the builder of the output stream, call setDataCallback() with this stream.
+ * For the builder of the output stream, call setDataCallback() with this object.
  *
- * When both streams are ready, onAudioReady() will call onBothStreamsReady().
+ * When both streams are ready, onAudioReady() of the output stream will call onBothStreamsReady().
  * Callers must override onBothStreamsReady().
  *
  * To ensure best results, open an output stream before the input stream.
- * Call inputBuilder.setBufferCapacityInFrames(mOutputStream->getBufferSizeInFrames() * 2).
+ * Call inputBuilder.setBufferCapacityInFrames(mOutputStream->getBufferCapacityInFrames() * 2).
+ * Also, call inputBuilder.setSampleRate(mOutputStream->getSampleRate()).
  *
  * Callers must call setInputStream() and setOutputStream().
  * Call start() to start both streams and stop() to stop both streams.
+ * Caller is responsible for closing both streams.
+ *
+ * Callers should handle error callbacks with setErrorCallback() for the output stream.
+ * When an error callback occurs for the output stream, the caller should restart both streams.
```

Comment:
Tell them Oboe will close the output stream but they need to close the input stream before reopening both streams.
Also the caller needs to "reopen and restart" not just "restart".

Reply:
Thanks! I updated the comment.